main.js

```diff
@@ -31,3 +31,298 @@ mobileIcons.addEventListener('click', toggleMenuOptions);
+          </div>
+        </div>
+`;
+
+  multiStories.innerHTML += story;
+});
+
+function displayModal() {
+  const storyBtns = document.querySelectorAll('.btn');
+  storyBtns.forEach((storyBtn) => {
+    storyBtn.addEventListener('click', () => {
+      const btnId = storyBtn.getAttribute('id');
+      const storyInfo = stories[btnId];
+      if (window.innerWidth >= 768) {
+        let technologies = '';
+        storyInfo.storyDetails.technologies.map((technology) => {
+          technologies += `<li>${technology}</li>`;
+          return null;
+        });
+        const modalTemplate = `
+            <div class="modal">
+      <div class="desktop-modal-content">
+        <i class="close-modal-icon fa-solid fa-xmark fa-xl"></i>
+        <img src="${storyInfo.storyDetails.featuredImage}" class="img" alt="" />
+        <div class="below">
+          <h3> ${storyInfo.storyDetails.name} </h3>
+            <a href="${storyInfo.storyDetails.liveVersionLink}">See Live <img src="./images/live.svg" alt="" /></a>
+            <a href="${storyInfo.storyDetails.sourceLink}">See Source <img src="./images/source.svg" alt="" /></a>
+        </div>
+        <ul> ${technologies} </ul>
+        <p> ${storyInfo.storyDetails.description} </p>
+      </div>
+    </div>
+        `;
+        if (modalContainer.classList.contains('hide-modal')) {
+          modalContainer.classList.remove('hide-modal');
+          modalContainer.innerHTML = modalTemplate;
+        }
+        closePopup();
+      } else {
+        // logic for mobile version
+        let technologies = '';
+        storyInfo.mobileStoryDetails.technologies.map((technology) => {
+          technologies += `<li>${technology}</li>`;
+          return null;
+        });
+        const mobileModalTemplate = `
+          <div class="mobile-modal">
+            <div class="mobile-modal-content">
+              <i class="close-mobile-modal-icon fa-solid fa-xmark fa-xl"></i>
+              <img src="${storyInfo.mobileStoryDetails.featuredImage}" class="img" alt="" />
+                <div class="below">
+                  <h3> ${storyInfo.mobileStoryDetails.name} </h3>
+                  <ul> ${technologies} </ul>
+                  <p> ${storyInfo.mobileStoryDetails.description} </p>
+                  <div class="links">
+                    <a href="${storyInfo.mobileStoryDetails.liveVersionLink}">See Live <img src="./images/live.svg" alt="" /></a>
+                    <a href="${storyInfo.mobileStoryDetails.sourceLink}">See Source <img src="./images/source.svg" alt="" /></a>
+                  </div>
+                </div>
+            </div>
+          </div>
+        `;
+        if (mobileModalContainer.classList.contains('hide-modal')) {
+          mobileModalContainer.classList.remove('hide-modal');
+          mobileModalContainer.innerHTML = mobileModalTemplate;
+        }
+        closeMobilePopup();
+      }
+    });
+  });
+}
+
+displayModal();
```

Comment:
- [ ] **Required:** You have done an amazing job so far. 👏 
However, I noticed the following that should be fixed.

- The desktop popup is good. but the mobile needs some spacing.
- You could reduce the height of the modal, and picture and truncate the text to fit a fixed length irrespective of the amount of text received.
- Dim the main page to give attention to the popup on mobile.